madmom/evaluation/key.py

```diff
@@ -0,0 +1,246 @@
+    else:
+        return 'other', 0.0
+
+
+class KeyEvaluation(EvaluationMixin):
+    """
+    Provide the key evaluation score.
+
+    Parameters
+    ----------
+    detection : str
+        File containing detected key
+    annotation : str
+        File containing annotated key
+    strict_fifth : bool
+        Use strict interpretation of the 'fifth' category, as in MIREX.
+    name : str, optional
+        Name of the evaluation object (e.g., the name of the song).
+
+    """
+
+    METRIC_NAMES = [
+        ('score', 'Score'),
+        ('error_category', 'Error Category')
+    ]
+
+    def __init__(self, detection, annotation,
+                 strict_fifth=False, name=None, **kwargs):
+        self.name = name or ''
+        self.detection = key_label_to_class(open(detection).read().strip())
+        self.annotation = key_label_to_class(open(annotation).read().strip())
+        err_cat, score = error_type(self.annotation, self.detection,
+                                    strict_fifth)
+        self._category = err_cat
+        self._score = score
+
+    @property
+    def score(self):
+        """Evaluation score."""
+        return self._score
+
+    @property
+    def error_category(self):
+        """Error category."""
+        return self._category
+
+    def tostring(self, **kwargs):
+        """
+        Format the evaluation as a human readable string.
+
+        Returns
+        -------
+        str
+            Evaluation score and category as a human readable string.
+
+        """
+        return '{}: {:3.1f}, {}'.format(
+            self.name, self.score, self.error_category)
+
+
+class KeyMeanEvaluation(EvaluationMixin):
+    """
+    Class for averaging key evaluations.
+
+    Parameters
+    ----------
+    eval_objects : list
+        Key evaluation objects.
+    name : str, optional
+        Name to be displayed.
+
+    """
+
+    METRIC_NAMES = [
+        ('correct', 'Correct'),
+        ('fifth', 'Fifth'),
+        ('relative', 'Relative'),
+        ('parallel', 'Parallel'),
+        ('other', 'Other'),
+        ('weighted', 'Weighted'),
+    ]
+
+    def __init__(self, eval_objects, name=None):
+        self.name = name or 'mean for {:d} files'.format(len(eval_objects))
+
+        n = len(eval_objects)
+        c = Counter(e.error_category for e in eval_objects)
+
+        self.correct = float(c['correct']) / n
+        self.fifth = float(c['fifth']) / n
+        self.relative = float(c['relative']) / n
+        self.parallel = float(c['parallel']) / n
+        self.other = float(c['other']) / n
+        self.weighted = sum(e.score for e in eval_objects) / n
+
+    def tostring(self, **kwargs):
+        return ('{}\n  w: {:5.2f}  c: {:5.2f}  f: {:5.2f}, '
+                'r: {:5.2f}  p: {:5.2f}  o: {:5.2f}'.format(
+                    self.name,
+                    self.weighted * 100, self.correct * 100, self.fifth * 100,
+                    self.relative * 100, self.parallel * 100, self.other * 100)
```

Comment:
I'd prefer simple 0.x values rather than having them as %-values. Also `KeyEvaluation` has "normal" scores with range 0..1. But this is your decision.

Put this aside, the returned string does 1) not render the newline (at least not on my system), 2) is not really descriptive, 3) doesn't display that the values are in %. In short: I'd expect this function to be a bit more human readable 😏 .

Reply:
In the other evals, TP/FP/TN/FN are usually reported `%5.2`, other numbers `%.3`. I don't have a strong opinion on this, however, `%5.2` packs more detail into 5 characters than `%.3` (e.g. 87.32 vs 0.873). I'd probably favour consistency, though.

Reply:
BTW, same is true for chord eval, where I report `value * 100`